store/tikv/region_cache.go

```diff
@@ -1642,18 +1795,84 @@ const (
 	unreachable
 )
 
+func (s *Store) startHealthCheckLoopIfNeeded(c *RegionCache) {
+	// This mechanism doesn't support non-TiKV stores currently.
+	if s.storeType != TiKV {
+		logutil.BgLogger().Info("[health check] skip running health check loop for non-tikv store",
+			zap.Uint64("storeID", s.storeID), zap.String("addr", s.addr))
+		return
+	}
+
+	// It may be already started by another thread.
+	if atomic.CompareAndSwapInt32(&s.needForwarding, 0, 1) {
+		go s.checkUntilHealth(c)
+	}
+}
+
+func (s *Store) checkUntilHealth(c *RegionCache) {
+	ticker := time.NewTicker(time.Millisecond * 500)
```

Comment:
Is it too frequent, what's the old value?

Reply:
It's new code and there's no old value. I think I'll try to increase it to 1s. Is it ok?

Reply:
OKay, and notice the log printing, avoiding too many annoying logs that quickly exhaust the disk.